### PR TITLE
chore: add more log entries to understand issue of missing loan throu…

### PR DESCRIPTION
**Pull request by philipstanislaus**
…gh handleShelfIssue, simplify saving logic

### CODE REVIEW

**Comment by philipstanislaus** [comment]
Deployed to: https://thegraph.com/explorer/subgraph/centrifuge/tinlake-v3?version=pending and to our own hosted graph node: https://thegraph.centrifuge.io/subgraphs/name/centrifuge/tinlake

File: src/mappings/Shelf.ts [hunk 1]
```diff
@@ -25,17 +25,17 @@ export function handleShelfIssue(call: IssueCall): void {
   let loanId = loanIdFromPoolIdAndIndex(poolId, loanIndex)
 
   let pool = Pool.load(poolId)
-  let poolChanged = false
 
   if (!pool.loans.includes(poolId)) {
```
Comment (offerijns) [inline src/mappings/Shelf.ts:29]: I think this should be:
```suggestion
  if (!pool.loans.includes(loanId)) {
```

Comment (philipstanislaus) [reply src/mappings/Shelf.ts:29]: Happy to merge it into this branch as a next step 👍

File: src/mappings/Shelf.ts [hunk 2]
```diff
@@ -25,17 +25,17 @@ export function handleShelfIssue(call: IssueCall): void {
...
     // TODO: maybe optimize by using a binary search on a sorted array instead
     log.debug('handleShelfIssue: will add loan {} to pool {}', [loanId, poolId])
+    let l1 = pool.loans.length
```
Comment (offerijns) [inline src/mappings/Shelf.ts:32]: If everything works, I think this can be removed again?

File: src/mappings/Shelf.ts (hunk 1 above)
Comment (offerijns) [reply src/mappings/Shelf.ts:29]: I think you can merge this in now right?

**Review by offerijns — APPROVED** [review]
Looks great!

File: src/mappings/Shelf.ts (hunk 2 above)
Comment (philipstanislaus) [reply src/mappings/Shelf.ts]: Done

File: src/mappings/Shelf.ts (hunk 1 above)
Comment (philipstanislaus) [reply src/mappings/Shelf.ts]: Done in 
a5ff0e3

**Pull request merged**